Variables cannot defined inside a for definition

Let`s look for this C code:
```c
#include <stdio.h>
void g()
{
        for (int i=0;i<10;i++){
                printf("%d ",i);
        }
}
int main(){
        g();
        return 0;
}
```
Result `gcc`:
```
`○ → gcc -o part part.c; ./part
0 1 2 3 4 5 6 7 8 9 
```

Go code:
```golang
package main

import "io/ioutil"
import "testing"
import "os"
import "github.com/elliotchance/c2go/noarch"

//.....
func g() int {
        for nil; i < 10; func() int {
                i += 1
                return i
        }() {
                noarch.Printf([]byte("%d \x00"), i)
        }
}
func main() {
        __init()
        g()
        return
}
```

```cmd
○ → go run part.go
# command-line-arguments
./part.go:109:6: nil evaluated but not used
./part.go:109:11: undefined: i
./part.go:110:3: undefined: i
./part.go:111:10: undefined: i
./part.go:113:36: undefined: i
./part.go:115:1: missing return at end of function
```

initialization of var `i` is lost.

--- a/tests/for.c
+++ b/tests/for.c
@@ -3,7 +3,7 @@
 
 int main()
 {
-    plan(33);
+    plan(40);
 
     int i = 0;
 
@@ -79,6 +79,16 @@ int main()
 	diag("Without body and with 2 and more conditions");
 	for(i = 0, j = 0; j++,i++,i < 2; );
 	pass("%d",i)
+
+	diag("initialization inside for");
+	for(int h = 0;h < 3; h++)
+		pass("%d",h)
+	
+	diag("double initialization inside for");
+	for(int f = 0 , g = 0;g < 2 || f < 2; g++, f++){
+		pass("%d",f)
+		pass("%d",g)
+	}
 
     done_testing();
 }

--- a/transpiler/branch.go
+++ b/transpiler/branch.go
@@ -131,17 +131,17 @@ func transpileForStmt(n *ast.ForStmt, p *program.Program) (
 		panic("non-nil child 1 in ForStmt")
 	}
 
-	// If we have 2 and more initializations like
-	// in operator for
-	// for( a = 0, b = 0, c = 0; a < 5; a ++)
 	switch c := children[0].(type) {
 	case *ast.BinaryOperator:
 		if c.Operator == "," {
+			// If we have 2 and more initializations like
+			// in operator for
+			// for( a = 0, b = 0, c = 0; a < 5; a ++)
 			// recursive action to code like that:
 			// a = 0;
 			// b = 0;
 			// for(c = 0 ; a < 5 ; a++)
-			before, newPre, newPost, err := transpileToStmt(c.Children()[0], p)
+			before, newPre, newPost, err := transpileToStmt(children[0], p)
 			if err != nil {
 				return nil, nil, nil, err
 			}
@@ -149,6 +149,18 @@ func transpileForStmt(n *ast.ForStmt, p *program.Program) (
 			preStmts = append(preStmts, before)
 			preStmts = append(preStmts, newPost...)
 			children[0] = c.Children()[1]
+		}
+	case *ast.DeclStmt:
+		{
+			// If we have 2 and more initializations like
+			// in operator for
+			// for(int a = 0, b = 0, c = 0; a < 5; a ++)
+			newPre, err := transpileToStmts(children[0], p)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			children[0] = nil
+			preStmts = append(preStmts, newPre...)
 		}
 	}
 

